spacy/schemas.py

```diff
@@ -329,6 +329,7 @@ class ConfigSchemaTraining(BaseModel):
     frozen_components: List[str] = Field(..., title="Pipeline components that shouldn't be updated during training")
     annotating_components: List[str] = Field(..., title="Pipeline components that should set annotations during training")
     before_to_disk: Optional[Callable[["Language"], "Language"]] = Field(..., title="Optional callback to modify nlp object after training, before it's saved to disk")
+    before_update: Optional[Callable[["Language", int], None]] = Field(..., title="Optional callback that is invoked at the start of each training step")
```

Comment:
We might want to consider making this signature forward-compatible, i.e., use a `Dict[str, Any]` as the second parameter to allow for passing more values to the callback in the future.

Reply:
It's extremely hard to change the config setting types in the future (other than widening), so it would make sense to go ahead and implement the more general version, I think?